.github/workflows/lint.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -eu
+
+declare -a scan_files
+
+# Scan any "*.sh" and "*.bash" files.
+while IFS= read -r -d '' file; do
+    scan_files+=("$file")
+done < <(find . -type f \( -name '*.sh' -o -name '*.bash' \) -print0)
+
+perm_executable="/111"
+# BSD `find` on macOS does not support the slash syntax for permissions.
+[ "$(uname -s)" = "Darwin" ] && perm_executable="+111"
+
+# Scan any executable files that start with a shebang.
+while IFS= read -r -d '' file; do
+    [ "$(head -c2 "$file")" = '#!' ] || continue
+    scan_files+=("$file")
+done < <(find . -type f ! -name '*.*' -perm "$perm_executable" -print0)
+
+# Convert shellcheck "gcc" format to GitHub Actions annotations.
+# https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-error-message
+annotate_shellcheck() {
+    local file lineno _ severity message
+    while IFS=: read -r file lineno _ severity message; do
+        case "${severity# }" in
+        error ) severity="error" ;;
+        warning ) severity="warning" ;;
+        info | note | style ) severity="notice" ;;
+        esac
+        printf "::%s file=%s,line=%d::%s\n" "$severity" "$file" "$lineno" "${message# }"
+    done
```

Comment:
The proposed solution of passing shellcheck output straight to `jq` is cute and would work in some other scenarios, but in this case I need to translate the severity type from shellcheck terminology to GitHub Actions, and for now I'd rather do that in bash `case` statement rather than in jq.

Reply:
Okay, I've looked up how to do the translation using jq and I've implemented the suggestion now. Thanks!